packages/block-editor-tools/src/hooks/use-option/index.js

```diff
@@ -0,0 +1,34 @@
+import { useSelect, useDispatch } from '@wordpress/data';
+
+export default function useOption(name) {
+  const selectors = useSelect((select) => {
+    const { getEditedEntityRecord, getEntityRecordEdits, isSavingEntityRecord } = select('core');
+    const settings = getEditedEntityRecord('root', 'site');
+    const edits = getEntityRecordEdits('root', 'site');
+    return {
+      value: settings?.[name],
+      isEdited: !!edits?.[name],
+      isSaving: isSavingEntityRecord('root', 'site'),
+    };
+  }, []);
+
+  const { editEntityRecord, saveEntityRecord } = useDispatch('core');
+
+  function onChange(newValue) {
+    editEntityRecord('root', 'site', undefined, {
+      [name]: newValue,
+    });
+  }
+
+  async function onSave() {
+    await saveEntityRecord('root', 'site', {
+      [name]: selectors.value,
+    });
+  }
+
+  return {
+    ...selectors,
+    onChange,
+    onSave,
+  };
+}
```

Comment:
Any reason not to convert this file to Typescript? This is a goal of this project: https://github.com/alleyinteractive/alley-scripts/issues/106 

Docblocks would be helpful as well.

Reply:
The reason was that I didn't know we had TS support even enabled, I saw that everything else was JS and rolled with it. Thanks for calling that out, I updated the hook!